### PR TITLE
Introduce index_dropped check

### PR DESCRIPTION
**Pull request by kamilkowalski**
This PR adds a new check to detect when indexes are dropped in migrations. Dropping indexes can be dangerous in production as it can impact query performance and potentially cause full table scans. The check warns developers when indexes are dropped so they can carefully consider the implications.

It also cleans up `Runner` tests - there was a duplicate test case verifying valid files, and with this new check migrations that roll back index creation are also marked as dangerous so I changed the structure a bit and removed one test migration.

### CODE REVIEW

File: test/ast_parser_test.exs [hunk 1]
```diff
@@ -358,6 +358,18 @@ defmodule ExcellentMigrations.AstParserTest do
     assert [table_dropped: 1] == AstParser.parse(ast4)
   end
 
+  test "detects index dropped" do
+    ast1 = string_to_ast("drop index(:recipes, [:cuisine], concurrently: true)")
```
Comment (Artur-Sulej) [inline test/ast_parser_test.exs:362]: I propose also testing version without any options, because it's a different arity that produces slightly different AST structure. 
```suggestion
    ast = string_to_ast("drop index(:recipes, [:cuisine])")
    ast1 = string_to_ast("drop index(:recipes, [:cuisine], concurrently: true)")
```

File: test/example_migrations/20220726000151_create_index_concurrently_valid.exs [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-    create(unique_index(:dumplings, [:dough], concurrently: true))
-    create_if_not_exists(unique_index(:dumplings, [:dough], concurrently: true))
-  end
-
-  def down do
-    drop(index(:dumplings, [:dough], concurrently: true))
-    drop_if_exists(index(:dumplings, [:dough], concurrently: true))
-    drop(unique_index(:dumplings, [:dough], concurrently: true))
-    drop_if_exists(unique_index(:dumplings, [:dough], concurrently: true))
-  end
```
Comment (Artur-Sulej) [inline test/example_migrations/20220726000151_create_index_concurrently_valid.exs]: One thing that comes to my mind now when looking at this change is that dropping index could be considered unsafe only if done in `up` or `change`, while in `down` it's actually fine.

Comment (katafrakt) [reply test/example_migrations/20220726000151_create_index_concurrently_valid.exs]: I think it generally would be great to disable checking `down`. On multiple occasions I have a migration using `change` that passes the check, then rewrite it to use `up`/`down` and the checks are triggered.

We never roll back migrations in production (similar to many teams, I guess) so checking this part of the migration could be skipped without consequences.
